src/rank_llm/rerank/rank_listwise_os_llm.py

```diff
@@ -105,7 +108,7 @@ def create_prompt(
     ) -> Tuple[str, int]:
         query = retrieved_result["query"]
         num = len(retrieved_result["hits"][rank_start:rank_end])
-        max_length = 300
+        max_length = 300 * (20 / (rank_end - rank_start))
```

Comment:
ditto

Reply:
resolving, since we will solve this in a follow up to make sure reported results are fine.